athena-federation-sdk/src/test/java/com/amazonaws/athena/connector/lambda/handlers/GlueMetadataHandlerTest.java

```diff
@@ -214,7 +215,8 @@ public void doListTables()
                     return mockResult;
                 });
 
-        ListTablesRequest req = new ListTablesRequest(IdentityUtil.fakeIdentity(), queryId, catalog, schema);
+        ListTablesRequest req = new ListTablesRequest(IdentityUtil.fakeIdentity(),
+                queryId, catalog, schema, null, NULL_PAGE_SIZE);
```

Comment:
i think we should make this test paginated

Reply:
GlueMetadataHandler.doListTables cannot be paginated:

Reply:
The specific connectors' implementation of `doListTables` calls `GlueMetadataHander.doListTables` to obtain the list of tables from Glue, then adds additional tables from the actual data source. When a list is an aggregation from more than one data source, pagination based solely on one of the lists, would undoubtedly create duplication of tables. The separate lists must be combined first then paginated. Therefore, pagination should take place in the actual connectors' `doListTables` implementation after the aggregation yields a full list of tables.

Reply:
Added full pagination support for the GlueMetadataHandler.doListTables API and added tests.